### ISSUE TITLE
Should amp.initialize be called on model not having it's weights updated

### ISSUE THREAD

**Issue by Genius1237** [NVIDIA/apex#898]
I'm performing training where I have 2 copies of a model
- The main model, the one on which the gradients are used to make weight updates
- A second model which is used in calculating the loss function, but does not have it's weights updated.

My crude version of my code looks something like this
```
stock_model = copy.deepcopy(model)
model, optimizer = amp.initialize(model, optimizer, opt_level=args.fp16_opt_level)
.
.
for _ in range(num_epochs):
    for x in dataloader:
        out = model(x)
        with torch.no_grad():
            out_stock = model(x)
        loss = loss_fn(out, out_stock)

        with amp.scale_loss(loss, optimizer) as scaled_loss:
            scaled_loss.backward()
        
        optimizer.step()
        model.zero_grad()
```

As you can see, `stock_model` is used in the loss calculation, but it's weights aren't updated. Should it be sent to `amp.initialize` as well?
